docs/docs/3-Files.md

```diff
@@ -30,26 +30,43 @@ following files.
 
 === "`config.toml` file for the configuration example"
     ```toml
+    # Config for an example
+
     # Use this file to set the config that you prefer by editing the file
-    name = "application specific name"
+
+    # The next element is refering to the Config container class that
+    # defines the config parameters of the application.
+    # It is used when loading from command line
+    Config_container_class = "config.MyExampleConfig"
+
+
+    # With this parameter you can configure the name; defaults to 'nice example'
+    name = "the real thing"
+
+    # Holds the configuration parameters for the first section
     [section1]
-    # field1 has default value 0.5
+
+    # The first field; defaults to 0.5
     field1 = -0.5
-    # field2 has default value 2
-    field2 = 22
 
+    # The second field; defaults to 2
+    field2 = 22
     ```
 
 === "`settings.json` file for the settings example"
     ```json
     {
+        "Settings_container_class": "settings.MyExampleSettings",
```

Comment:
Could we use something like `__Settings_container_class__`, the name is already pretty unique, but we could signal that it's a 'special' type with the underscores?

Reply:
I agree, done. I also created a single method that renders the key from the parameter kind. And then also introduced a ParameterKind, so that we are relying less on a particular string value. As a result, lots of files have changed, sorry for that. It is better now, I think.